src/state/reducer.ts

```diff
@@ -21,7 +21,7 @@ export const reducer = (state: GlobalState, action: Action): GlobalState => {
     case 'setArnsContractState':
       return {
         ...state,
-        arnsSourceContract: action.payload,
+        arnsSourceContract: action.payload as ArNSContractState,
```

Comment:
i don't think this cast is needed with the change from how we're dispatching now (the full state)